[FIX] web, *: Update X/Twitter logo

xlu-odoo opened the pull request:
*: website, website_twitter

This commit introduces an update to the Font Awesome library, specifically replacing the X/Twitter icon with its new design. It also involves adjustments to a few images and SVG files to incorporate the updated X/Twitter logo.

Technical note: It updates an existing file, originally modified for the addition of the TikTok icon, which was not available in Font Awesome 4.

Related to: https://github.com/odoo/enterprise/pull/53644

task-3463530



---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/148126).

jcs-odoo commented:
Hello @xlu-odoo 

In the footer of the Website, the X icon on a white square doesn't work well: the stroke is too thin compared to the other icons. In my opinion, it should rather be a white stroke with no background, like Instagram, as the square shape background is not part of the logo design as it is the case for Facebook and LinkedIn
<img width="176" alt="image" src="https://github.com/odoo/odoo/assets/46810843/d1d352bc-5327-44f6-adb0-e0549c8f8fcd">

For the app icon, is there a reason to choose the white-on-black logo? Considering that the black background is not a shape, but just a color, using the "positive" version may be more appropriate. Also, this icon doesn't follow the existing design (in 15.0) as the current Facebook and LinkedIn icons (but neither does Instagram)

Somehow, for the Social Marketing feeds, the black background seems more appropriate, but again, the stroke is too thin. Not sure if this can be improved, though.

**I wonder if X has a dedicated "small icon" version of their logo, though.** When I look at the favicons of X and Twitter, the stroke difference doesn't look this bad. The padding is smaller than in our assets, though.
<img width="176" alt="image" src="https://github.com/odoo/odoo/assets/46810843/23843475-b049-439d-97ce-76ad37a1c996">

Additionally, it's weird not to update the text as well: reading "Twitter" everywhere doesn't make any sense with this logo, now.

I hope this helps :)

xlu-odoo commented:
Hello @jcs-odoo , thanks a lot for the feedbacks.

> In the footer of the Website, the X icon on a white square doesn't work well: the stroke is too thin compared to the other icons. In my opinion, it should rather be a white stroke with no background, like Instagram, as the square shape background is not part of the logo design as it is the case for Facebook and LinkedIn <img alt="image" width="176" src="https://private-user-images.githubusercontent.com/46810843/295268729-d1d352bc-5327-44f6-adb0-e0549c8f8fcd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ5NjA2NzIsIm5iZiI6MTcwNDk2MDM3MiwicGF0aCI6Ii80NjgxMDg0My8yOTUyNjg3MjktZDFkMzUyYmMtNTMyNy00NGY2LWFkYjAtZTA1NDljOGY4ZmNkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTExVDA4MDYxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThkMGY3YjA5ZmQyZjYxM2FmY2M4NWY0OTNhNjgzOTc5NWJmMTA4NGYwZjZkODEyY2YyMWVlN2NlYmY3OTFhMTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Vq4_ovRs9MEx6FyeqNhkBGFJVsjI45MBUFXabcHlioQ">

Replaced with just the stroke version (thicker version).

> For the app icon, is there a reason to choose the white-on-black logo? Considering that the black background is not a shape, but just a color, using the "positive" version may be more appropriate. Also, this icon doesn't follow the existing design (in 15.0) as the current Facebook and LinkedIn icons (but neither does Instagram)

Updated with black-on-white. 


> Somehow, for the Social Marketing feeds, the black background seems more appropriate, but again, the stroke is too thin. Not sure if this can be improved, though. I wonder if X has a dedicated "small icon" version of their logo, though.** When I look at the favicons of X and Twitter, the stroke difference doesn't look this bad. The padding is smaller than in our assets, though. <img alt="image" width="176" src="https://private-user-images.githubusercontent.com/46810843/295271424-23843475-b049-439d-97ce-76ad37a1c996.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ5NjA2NzIsIm5iZiI6MTcwNDk2MDM3MiwicGF0aCI6Ii80NjgxMDg0My8yOTUyNzE0MjQtMjM4NDM0NzUtYjA0OS00MzlkLTk3Y2UtNzZhZDM3YTFjOTk2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTExVDA4MDYxMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTkzZTRhZmQ1MjEzNmNlZTRmODllMzkxYjIwODE3MjA5ZjU3OTI5YWI3YmZmNGU3MDNkYzExNGY3ODFkZmMzMzEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.W9uBu8mkjib_cMCxUZcq5Er3o8zEFVsA1uNYvP2a724">

Has been updated with their thicker version.


> Additionally, it's weird not to update the text as well: reading "Twitter" everywhere doesn't make any sense with this logo, now.

 Separate commit, started to replace twitter/tweet/retweet/RT,... references that should no longer exist.

jcs-odoo commented:
@xlu-odoo it looks better indeed :)

xlu-odoo commented:
Updated, thank you @anso-odoo . In addition to your found relics, I've found some more.

xlu-odoo commented:
@stefanorigano can you check this one when you have time?

euvm-odoo commented:
Hello. Is this being worked on? Have found several tickets from customers regarding this. Also, please let me know if there is anything I can help with :))

xlu-odoo commented:
> Hello. Is this being worked on? Have found several tickets from customers regarding this. Also, please let me know if there is anything I can help with :))

Hello! Yes it is, it just needs an additional look from @stefanorigano.

xlu-odoo commented on addons/website_twitter/__manifest__.py:
```diff
@@ -2,12 +2,12 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
 {
-    'name': 'Twitter Snippet',
```
Being cautious, I now left some `Twitter` terms/strings in some models and controllers (.py) (error message, module name etc,..)

rdeodoo reviewed (changes requested):
Why is it not targeting master (or 17.0 at worst)?
In stable:
- I'd only keep the font-awesome css change so the fa-twitter get the new icon and new color if you really want (I'd not do that in 15 and 16 tho)
- I'd remove all the wording / title / description change, as:
  - It will break translations
  - It will confuse user, eg changing a module name
  
Whatever happen with the PR, don't forget to update `.pot`

xlu-odoo commented:
> Why is it not targeting master (or 17.0 at worst)? In stable:
> 
> * I'd only keep the font-awesome css change so the fa-twitter get the new icon and new color if you really want (I'd not do that in 15 and 16 tho)
> * I'd remove all the wording / title / description change, as:
>   
>   * It will break translations
>   * It will confuse user, eg changing a module name
> 
> Whatever happen with the PR, don't forget to update `.pot`

Thank you for your feedback Romain! 

The original spec aimed to affect supported versions from 15 to the latest. 
Following your suggestions, the last update pushed (com & ent) now limit changes to images (except module icon for 15) / icon (fa) / color (except in website as asked for 15.)

If it seems good for you now, could you eventually merge so we can move forward wit others versions? 

( cc @stefanorigano ; I also added a new fontface for twitter, and revert original tiktok only one)

euvm-odoo commented:
Hello @xlu-odoo is the plan for this to target 15.0+??

xlu-odoo commented on addons/website/views/website_templates.xml:
```diff
@@ -1553,7 +1553,7 @@
                             <ul class="list-unstyled">
                                 <li class="py-1"><i class="fa fa-1x fa-fw fa-facebook-square mr-2"/><a href="/website/social/facebook" target="_blank">Facebook</a></li>
-                                <li class="py-1"><i class="fa fa-1x fa-fw fa-twitter-square mr-2"/><a href="/website/social/twitter" target="_blank">Twitter</a></li>
+                                <li class="py-1"><i class="fa fa-1x fa-fw fa-twitter mr-2"/><a href="/website/social/twitter" target="_blank">Twitter</a></li>
```
It shall be adapted for 17+

rdeodoo commented on addons/website/views/website_templates.xml:
```diff
@@ -1553,7 +1553,7 @@
                             <ul class="list-unstyled">
                                 <li class="py-1"><i class="fa fa-1x fa-fw fa-facebook-square mr-2"/><a href="/website/social/facebook" target="_blank">Facebook</a></li>
-                                <li class="py-1"><i class="fa fa-1x fa-fw fa-twitter-square mr-2"/><a href="/website/social/twitter" target="_blank">Twitter</a></li>
+                                <li class="py-1"><i class="fa fa-1x fa-fw fa-twitter mr-2"/><a href="/website/social/twitter" target="_blank">Twitter</a></li>
```
I'll remove this diff, the value is really low in stable, especially 15 (just a tiny bit more beautiful).
But the risk is not null, seems likely that someone xpath'd this element to add a custom social network after it or to hide it or whatever.

Not worth the risk in stable

rdeodoo reviewed (approved):
@robodoo r+

Thank you for having minimized the PR to the fa change and small preview images :+1: 
In master feel free to change everything :+1:

robodoo commented:
@xlu-odoo @rdeodoo linked pull request(s) odoo/enterprise#53644 not ready. Linked PRs are not staged until all of them are ready.

rdeodoo commented:
@robodoo r-
See enterprise PR

rdeodoo reviewed (approved):
@robodoo r+


I was hesitating a lot before merging this PR in stable. Even if the PR got trimmed down to the bare minimum.

But:
- as per SRI message [here](https://github.com/odoo/enterprise/pull/53644#issuecomment-1973164756), twitter name and old logo is no more a thing. Regardless if people like it or not: they rebranded.
- clients are complaining about not being able to easily replace the twitter outdated logo by new one, on the other hand, no one should complain about the logo update since it's.. the correct logo now that they rebranded
- we could tell people to upgrade to receive the full twitter -> x update, but it's obviously not a realistic solution for everyone
- the backend will still have some old logo images, which is not a real problem, the need is for the frontend to show the updated logo, people can live with the old logo in their internal backend
